TLP:AMBER hardcoded in email subject, issue #1107 on github.

cristianbell opened the pull request:
Added a new entry in the server settings set, updated necessary files.

cristianbell committed:
TLP:AMBER hardcoded in email subject, issue #1107 on github.
 Added a new entry in the server setting set, updated necessary files.

RichieB2B commented:
This PR adds app/Plugin/CakeResque/Config/config.php which is a configure file that can be modified by the site administrator. Especially on servers running multiple MISP instances. Please remove this file from the PR.

iglocska commented:
Hello Cristian,

the idea with the request was to instead check the attached tags on the event and see if a TLP tag is found - if yes, use that as the e-mail's TLP, if not revert to a default (which should ideally be based on the event's distribution level). Let me know if you have any questions!

cristianbell committed:
Add attribute under Events: type list is not alphabetical #1117.
 - changed the hardcoded $categoryDefinitions var in Attribute.php
 - minor change just added a asort($array) to the code in View\Attributes\add.ctp
 but this just a fix for the issue #1117 but not for the whole project, as the $categoryDefinitions is used in several places.

cristianbell commented:
Hi Andras, 
I know what the idea was, but I've spoken with @cvandeplas (who had the idea) before implementing any changes. 
He said that the best way forward is to add this to the server settings at the moment. Leave it like this for a while and see if a more automated approach is something that the user community really needs. 

:)

cristianbell committed:
trying to mend the config.php error add..

cristianbell committed:
mending the config.php add - part 2

cristianbell committed:
Free text import does not recognize new international Top Level domains #1149 - first push.

cristianbell committed:
just a few bug fixes:
- style="shortish" to class="shortish"
- added a missing curly brace "}" to main.css
- added .pointer class {cursor:pointer} in main.css

adulau commented on app/Controller/EventsController.php:
```diff
@@ -2763,8 +2763,17 @@ public function freeTextImport($id) {
 			App::uses('ComplexTypeTool', 'Tools');
+
+
+            Configure::write('debug', 2);
```
Enabling the debug is maybe something you would like to avoid in production ;-)

cristianbell replied:
;) sure, but this is not yet final. There was a part1 before, this would be part0, and part 2 and maybe part3 will follow. Atm I'm saving after each issue or when leaving the office.. :) 
I spoke w/ Andras several times and I'll let him know when it's in a state when it can be inspected..

was a quick fix for a local need, I'll do this in some local file which is never sent 2 github,:D

adulau replied:
Thank you for the feedback. So the best would be separated small and separated PR. It's easier for testing and review. Thank you and enjoy your evening.

cristianbell replied:
:) already spoke w/ Andras about this, I'll create separate branches for separate issues. 
And thanks, have a nice evening and weekend as well!

cristianbell committed:
Free text import does not recognize new international Top Level domains #1149 - second push.

cristianbell committed:
Merge branch '2.4' of https://github.com/MISP/MISP into 2.4

cristianbell committed:
Free text import does not recognize new international Top Level domains #1149 - thirs push.

cristianbell committed:
minor update, the http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm doesn't exist anymore, changed it to the current file: 7-6

Rafiot commented:
First of all, the PR has merge conflict with 2.4, we won't be able to merge it.

Beside that, here are the generic requirements in order to contribute to MISP:
- One Pull Request per fix/feature/change/... to avoid polluting the history: I see you're trying to fix at least two things (email subject and free text import)
- Keep the amount of commits per PR as small as possible: if for any reason, you need to fix your commit after the pull request, please squash the changes in one single commit (or tell us why not)
- I see you added your config file (containing GPG passwords and salt (!)). If we merge this branch, it will be in the history of the project and anyone can just grab it out, even if you deleted the file in a later commit.
- Always make sure it is mergeable in the default branch (as of today 2016-06-03: branch 2.4)
- Please make sure Travis CI works on this request, update the test cases if needed
- Any major changes adding a functionality should be disabled by default in the config

rotanid commented:
obsolete, replaced by #1240

Rafiot commented:
True, closing.
